components/vc/pdu/src/powerManager.c

```diff
@@ -81,33 +84,39 @@ static void powerManager_periodic_10Hz(void)
 
     for (uint8_t i = 0; i < DRV_TPS2HB16AB_IC_COUNT; i++)
     {
-        drv_tps2hb16ab_setEnabled(i, DRV_TPS2HB16AB_OUT_1, enable_loads);
-        drv_tps2hb16ab_setEnabled(i, DRV_TPS2HB16AB_OUT_2, enable_loads);
-        tmp_current += drv_tps2hb16ab_getCurrent(i, DRV_TPS2HB16AB_OUT_1);
-        tmp_current += drv_tps2hb16ab_getCurrent(i, DRV_TPS2HB16AB_OUT_2);
+        for (uint8_t n = 0; n < DRV_TPS2HB16AB_OUT_COUNT; n++)
+        {
+            drv_hsd_state_E state = drv_tps2hb16ab_getState(i, n);
+            if (state == DRV_HSD_STATE_OVERCURRENT)
+            {
+                // TODO: Set alarm
+                drv_tps2hb16ab_setEnabled(i, n, false);
+            }
+            else if (state == DRV_HSD_STATE_OFF)
```

Comment:
nit: could also be cleaner as a switch. That way you can list out all the states you don't currently care about

Reply:
Yea im leaving this open because there are technically a few more things I would like to check for in this logic, but Im trying to avoid a huge ass set of commits